solver/Cargo.toml

```diff
@@ -36,6 +36,7 @@ tokio = { version = "0.2", features =["macros", "rt-threaded", "time"] }
 tracing = "0.1"
 web3 = { version = "0.15", default-features = false, features = ["http-tls"] }
 itertools = "0.10"
+orderbook= { path = "../orderbook" }
```

Comment:
solver and orderbook should not depend on another. Instead all shared components should live in the shared crate. Maybe in a separate PR, could you move `orderbook/src/price_estimate.rs` to `shared/src/price_estimate.rs `?

Reply:
Yes, that was my feeling too, I'll do that.

Reply:
done in another PR, now merged into main and this one.